### PR TITLE
[pt2][quant] Enable _fold_conv_bn_qat with BN in eval mode

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114927

**Summary**
Draft a fix as discussed in https://github.com/pytorch/pytorch/pull/114547#discussion_r1411559586, since we don't want to impose any ordering requirements between `move_exported_model_to_eval` and `convert_pt2e`

**Test Plan**
```
python -u -m pytest -s -v test_quantize_pt2e_qat.py -k test_qat_resnet18
```

### CODE REVIEW

File: torch/ao/quantization/pt2e/qat_utils.py [hunk 1]
```diff
@@ -715,19 +717,20 @@ def _fold_conv_bn_qat_helper(
         [True, False],  # is_per_channel
         [True, False],  # has_bias
         [True, False],  # bias_is_quantized
+        [True, False],  # bn_is_training
```
Comment (andrewor14) [inline torch/ao/quantization/pt2e/qat_utils.py:720]: We don't need to do this because all matching uses `ignore_literals`, so this doesn't actually do anything

**Review by andrewor14 — APPROVED** [review]

File: torch/ao/quantization/pt2e/qat_utils.py [hunk 2]
```diff
@@ -218,7 +219,7 @@ def _quantized_qat_conv_bn_pattern(
         x = x / scale_factor.reshape(bias_shape)
         if has_bias:
             x = x + kwargs["conv_bias"].reshape(bias_shape)
-        x = F.batch_norm(x, bn_running_mean, bn_running_var, bn_weight, bn_bias, training=True, eps=bn_eps)
+        x = F.batch_norm(x, bn_running_mean, bn_running_var, bn_weight, bn_bias, training=bn_is_training, eps=bn_eps)
```
Comment (jerryzh168) [inline torch/ao/quantization/pt2e/qat_utils.py:222]: can you do ignore literals in matching and replacement?

Comment (leslie-fang-intel) [reply torch/ao/quantization/pt2e/qat_utils.py:222]: Hi @jerryzh168, thanks for the comment. Did you mean set `ignore_literals` as True in `replace_pattern_with_filters`. If so, I think it's already set as 
https://github.com/pytorch/pytorch/blob/77c4565d58fb9c82bd59ebbd5e22471cf8ecafe2/torch/ao/quantization/pt2e/qat_utils.py#L733-L740

However, here for `F.batch_norm` when `training` parameter is `True` or `False` actually generates 2 different aten operator after export as `torch.ops.aten._native_batch_norm_legit.default` and `torch.ops.aten._native_batch_norm_legit_no_training.default`. Will `ignore_literals` still work in this case?
